browser-sign-in/__tests__/App-test.js

```diff
@@ -1,12 +1,205 @@
-/**
- * @format
+/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
  */
 
- describe('app setup', () => {
-  it('is a dummy test', () => {
-    expect(true).toBe(true);
+import React from 'react';
+import App from '../App';
+import { shallow, render } from 'enzyme';
+import renderer from 'react-test-renderer';
+
+import { NativeModules, NativeEventEmitter } from 'react-native';
+
+jest.mock('NativeEventEmitter');
+
+const nativeEmitter = new NativeEventEmitter();
+
+jest.mock('NativeModules', () => {
+  return {
+    OktaSdkBridge: {
+      createConfig: jest.fn(),
+      signIn: jest.fn(),
+      signOut: jest.fn(),
+      getAccessToken: jest.fn(),
+      getIdToken: jest.fn(),
+      getUser: jest.fn(),
+      isAuthenticated: jest.fn(),
+      revokeAccessToken: jest.fn(),
+      revokeIdToken: jest.fn(),
+      revokeRefreshToken: jest.fn(),
+      introspectAccessToken: jest.fn(),
+      introspectIdToken: jest.fn(),
+      introspectRefreshToken: jest.fn(),
+      refreshTokens: jest.fn(),
+    }
+  }
+});
+
+global.fetch = jest
+  .fn()
+  .mockImplementation(() => {
+    const promise = new Promise((resolve, reject) => {
+      resolve({
+        json: () => {
+          return {
+            user: [{ foo: 'foo', bar: 'bar' }],
+          }
+        },
+        ok: true,
+      });
+    });
+    return promise;
+  })
+  .mockImplementationOnce(() => {
+    const promise = new Promise((resolve, reject) => {
+      resolve({
+        json: () => {
+          return {
+            userinfo_endpoint: 'dummy_endpoint',
+          }
+        },
+        ok: true,
+      });
+    });
+    return promise;
+  });
+
+describe('app setup', () => {
+  it('should render without crashing', () => {
+    const rendered = renderer.create(<App />).toJSON();
+    expect.assertions(1);
```

Comment:
You shouldn't have to put this every time, only when you have async tests that can't be automatically counted.  (and then I'd still suggest using the done() callback instead, because giving a number of assertions is brittle and risks future errors)

Reply:
deleted expect.assertions on none async functions